odak/wave/classical.py

```diff
@@ -237,28 +238,24 @@ def band_limited_angular_spectrum(field, k, distance, dx, wavelength):
     -------
     result           : np.complex
                        Final complex field (MxN).
+
     """
     nv, nu = field.shape
-    x = np.linspace(-nu/2*dx, nu/2*dx, nu)
-    y = np.linspace(-nv/2*dx, nv/2*dx, nv)
-    X, Y = np.meshgrid(x, y)
-    Z = X**2+Y**2
-    h = 1./(1j*wavelength*distance)*np.exp(1j*k*(distance+Z/2/distance))
-    h = np.fft.fft2(np.fft.fftshift(h))*dx**2
-    flimx = np.ceil(1/(((2*distance*(1./(nu)))**2+1)**0.5*wavelength))
-    flimy = np.ceil(1/(((2*distance*(1./(nv)))**2+1)**0.5*wavelength))
-    mask = np.zeros((nu, nv), dtype=np.complex64)
-    mask = (np.abs(X) < flimx) & (np.abs(Y) < flimy)
-    mask = set_amplitude(h, mask)
+    fx = np.linspace(-1. / 2. /dx, 1. /2. /dx, nu)
+    fy = np.linspace(-1. / 2. /dx, 1. /2. /dx, nv)
+    FX, FY = np.meshgrid(fx, fy)
+    H = np.exp(-1j * distance * (k - np.pi * wavelength * (FX**2 + FY**2) ))
     U1 = np.fft.fft2(np.fft.fftshift(field))
-    U2 = mask*U1
```

Comment:
I observe that with the new additions, this type of beam propagation generates a different result than before:

![odak_wave_Transfer_Function_Fresnel](https://github.com/user-attachments/assets/622be4c1-b367-44a8-b65b-f9bd3ed6fa3d)

I generated the above result by running `test/test_wave_propagate_beam.py` for distance of `5e-3`. On the other hand, the `Bandlimited Angular Spectrum` generates the following result when added to the list of `propagation_types` in that unit test:



![odak_wave_Bandlimited_Angular_Spectrum](https://github.com/user-attachments/assets/79cf9df3-a5ee-4115-b14e-43fe6741ff97)

Reply:
Thanks, I included the name now.

I'm not sure if I'm merging correctly. Now I removed the minus sign in propagation direction and test locally for 5e-3 m prop distance in test_wave_progate_beam.py. They look the same for transfer function kernel, impulse response kernel, bandlimited AS kernel and AS kernel. Please test again and let me know if it still not works for you, or if you prefer me to create a new pull request.

Transfer function kernel in wave.classical
![image](https://github.com/user-attachments/assets/cd73415f-4bfd-46b1-97e0-2a993f6dd165)

Impulse response kernel in wave.classical
![image](https://github.com/user-attachments/assets/6a10b2f0-d5b4-4e03-90dc-5b5c3f84df85)

Bandlimit AS kernel in wave.classical
![image](https://github.com/user-attachments/assets/2d997a31-fe99-495f-b046-30c55a1d0725)

AS kernel in wave.classical
![image](https://github.com/user-attachments/assets/7b9c3f47-12e4-4ded-a4e0-4677a52aa0ac)

Reply:
I can't reproduce the above results on my end. Could you please make sure to completely remove `odak` and install the version that you have updated from your repository?

If you are using `pip` for installation, please first make sure that you have removed it from your local by typing `pip3 uninstall odak` . In some cases, people use `sudo` in their installation. To make double sure that you have completely, removed it. Please make sure to type `sudo pip3 uninstall odak`.

To install it using your local version, navigate to the directory that you have your `odak` using a terminal (e.g., `cd PATH/odak`). Once you are at the root directory, simply type `pip3 install -e .` to install your version.

Please also revert the sign modification from `odak.learn.wave` as this would break many of our existing repositories.

Reply:
* I am confident that you are reproducing the above results using `odak` in `Github:kaanaksit/odak` as I can reproduce the same result as yours when I use that.

Reply:
Ah I see. Yes you are correct. I was using previous ``odak`` by mistake. Now I can see my version producing the error and they have been fixed now. 

I also revert the sign modification in propagation direction. Please test and let me know. Thanks.